cpp/apidoc/Doxyfile

```diff
@@ -1,21 +1,4 @@
-# Licensed to the Apache Software Foundation (ASF) under one
-# or more contributor license agreements.  See the NOTICE file
-# distributed with this work for additional information
-# regarding copyright ownership.  The ASF licenses this file
-# to you under the Apache License, Version 2.0 (the
-# "License"); you may not use this file except in compliance
-# with the License.  You may obtain a copy of the License at
-#
-#   http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing,
-# software distributed under the License is distributed on an
-# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-# KIND, either express or implied.  See the License for the
-# specific language governing permissions and limitations
-# under the License.
-
-# Doxyfile 1.8.13
+# Doxyfile 1.8.16
```

Comment:
@wesm Is this file autogenerated? I'm guessing we either need to add back the ASF header or add this to the 🐀 exclude file.

Reply:
Indeed: https://github.com/apache/arrow/pull/6557/checks?check_run_id=496256402#step:5:91